corim/signedcorim.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package corim
+
+import (
+	"crypto"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	cose "github.com/veraison/go-cose"
+)
+
+var (
+	ContentType = "application/rim+cbor"
+)
+
+// SignedCorim encodes a signed-corim type (i.e., a COSE Sign1 wrapped CoRIM)
+// with signature and verification methods
+type SignedCorim struct {
+	UnsignedCorim UnsignedCorim
+	message       *cose.Sign1Message
+}
+
+var (
+	algID       = cose.GetCommonHeaderTagOrPanic("alg")
+	contentType = cose.GetCommonHeaderTagOrPanic("content type")
+)
+
+func (o *SignedCorim) processHdrs() error {
+	var hdr = o.message.Headers
+
+	if hdr.Protected == nil {
+		return errors.New("missing mandatory protected header")
+	}
+
+	v, ok := hdr.Protected[contentType]
+	if !ok {
+		return errors.New("missing mandatory content type")
+	}
+
+	if v != ContentType {
+		return fmt.Errorf("expecting content type %q, got %q instead", ContentType, v)
```

Comment:
More of a question:

Everything below looks optional to me, so why we say kid is mandatory below ? Similar question for content_type ?
Generic_Headers = (
       ? 1 => int / tstr,  ; algorithm identifier
       ? 2 => [+label],    ; criticality
       ? 3 => tstr / int,  ; content type
       ? 4 => bstr,        ; key identifier
       ? 5 => bstr,        ; IV
       ? 6 => bstr,        ; Partial IV
       ? 7 => COSE_Signature / [+COSE_Signature] ; Counter signature

Reply:
That's just how CoRIM is formulated:
```
protected-signed-corim-header-map = {
  corim.alg-id => int
  corim.content-type => "application/rim+cbor"
  corim.issuer-key-id => bstr
  corim.meta => corim-meta-map
  * cose-label => cose-values 
}
```